framework/include/utils/RawValueFunctor.h

```diff
@@ -0,0 +1,116 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseFunctor.h"
+
+namespace Moose
+{
+template <typename T>
+class RawValueFunctor : public FunctorBase<T>
+{
+public:
+  using typename Moose::FunctorBase<T>::ValueType;
+  using typename Moose::FunctorBase<T>::GradientType;
+  using typename Moose::FunctorBase<T>::DotType;
+
+  RawValueFunctor(const FunctorBase<typename ADType<T>::type> & ad_functor)
+    : FunctorBase<T>(ad_functor.functorName() + "_raw_value", {EXEC_ALWAYS}),
+      _ad_functor(ad_functor)
+  {
+  }
+
+  virtual bool isExtrapolatedBoundaryFace(const FaceInfo & fi,
+                                          const Elem * const elem) const override
+  {
+    return _ad_functor.isExtrapolatedBoundaryFace(fi, elem);
+  }
+  virtual bool isConstant() const override { return _ad_functor.isConstant(); }
+  virtual bool hasBlocks(const SubdomainID id) const override { return _ad_functor.hasBlocks(id); }
+  virtual bool hasFaceSide(const FaceInfo & fi, const bool fi_elem_side) const override
+  {
+    return _ad_functor.hasFaceSide(fi, fi_elem_side);
+  }
+
+protected:
+  ///@{
+  /**
+   * Forward calls to wrapped object
+   */
+  ValueType evaluate(const ElemArg & elem, unsigned int state = 0) const override
```

Comment:
could we template all this

Reply:
What do you mean? The class is templated

Reply:
template methods cannot be virtual

Reply:
template on the functor arguments I meant

Reply:
ugh C++

Reply:
I don't think it's fair to critique C++ because it has two neat features that can't be intermingled 😄

Reply:
I could create a template method that all the virtual overrides call...but the virtual overrides would go from being one line to one line

Reply:
yeah not worth it